### ISSUE TITLE
'Some/Detect' ControlFlow

### ISSUE THREAD

**Issue by paraggupta1993** [caolan/async#591]
Is there a way to run 'n' asynchronous functions in parallel such that it will pass the callback with no error when say some ( or atleast 1 ) function didn't returned err.

Use Case :
1. Suppose, I am making 'n' API calls for the data to 'n' different servers. Even if one of them returned me the data and rest (n-1) returned me error, I should be able to use this data calling the 'final callback' with no-error. 
2. While making 'n' API calls, the-final callback is called immediately as soon as one function returned the data.

If there exists a way, Kindly tell. 

Thanks.

**Comment by aearly** [comment]
This seems similar to #568

**Comment by aearly** [comment]
Closing this in favor of #568.  Note that there is already `async.detect`, that exits when it receives a truthy value, but it doesn't follow node conventions exactly, so it could be cumbersome to use.
